ooni/nettests/blocking/web_connectivity.py

```diff
@@ -173,9 +177,20 @@ def setUp(self):
         self.report['tcp_connect'] = []
         self.report['control'] = {}
 
-        self.hostname = urlparse(self.input).netloc
-        if not self.hostname:
-            raise AbsentHostname('No hostname', self.input)
+        parseResult = urlparse(self.input)
+
+	if parseResult.scheme == "":
+            url = "http://{}".format(self.input)
+        elif parseResult.scheme in ['http', 'https']:
+            url = self.input
+        else:
+            raise InvalidScheme("Invalid Scheme", parseResult.scheme)
```

Comment:
Where should such a function reside? It could become part of a toolset to do things with urls in utils/url.py which I have already created. If you give your ok, I will make both instances use the suggested function.